ibis/expr/types/relations.py

```diff
@@ -769,7 +766,12 @@ def __getitem__(self, what):
             limit, offset = util.slice_to_limit_offset(what, self.count())
             return self.limit(limit, offset=offset)
 
-        values = tuple(bind(self, what, int_as_column=True))
+        args = [
+            self.columns[arg] if isinstance(arg, int) else arg
+            for arg in util.promote_list(what)
+        ]
+        values = self.bind(args)
```

Comment:
Am I understanding this right that `.__getitem__` accepts ints, but `.select` doesn't? This is a separate PR, but I would love to rip the bandaid off and just disallow ints everywhere.

Reply:
I think it's still useful to be able to do this. I use it during ad-hoc interactive analysis for referring to columns that have generated names.

Reply:
yeah I see what you're saying. I was trying to get these two methods to be totally equivalent, but I think I am convinced that the utility is worth the inconsistency.